From 4a251e26e447880a91a693afcba54212f8387450 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Wed, 18 Sep 2024 11:21:02 +0530
Subject: [PATCH] UPSTREAM: hostapd: Reject invalid Rejected Groups element in
 the parser There is no need to depend on all uses (i.e., both hostapd and
 wpa_supplicant) to verify that the length of the Rejected Groups field in the
 Rejected Groups element is valid (i.e., a multiple of two octets) since the
 common parser can reject the message when detecting this.

Tag: SAE
Patch-work: None
Patch-dependency: None

Signed-off-by: Jouni Malinen <j@w1.fi>
Signed-off-by: Arunpandi Kannan <quic_arunpand@quicinc.com>
---
 src/common/sae.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/common/sae.c b/src/common/sae.c
index e1ed323..73b5555 100644
--- a/src/common/sae.c
+++ b/src/common/sae.c
@@ -2123,6 +2123,12 @@ static int sae_parse_rejected_groups(struct sae_data *sae,
 		return WLAN_STATUS_UNSPECIFIED_FAILURE;
 	epos++; /* skip ext ID */
 	len--;
+	if (len & 1) {
+		wpa_printf(MSG_DEBUG,
+			   "SAE: Invalid length of the Rejected Groups element payload: %u",
+			   len);
+		return WLAN_STATUS_UNSPECIFIED_FAILURE;
+	}
 
 	wpabuf_free(sae->tmp->peer_rejected_groups);
 	sae->tmp->peer_rejected_groups = wpabuf_alloc(len);
-- 
2.34.1

